runtime/src/accounts_background_service.rs

```diff
@@ -49,15 +49,38 @@ pub type SnapshotRequestReceiver = Receiver<SnapshotRequest>;
 pub type DroppedSlotsSender = Sender<(Slot, BankId)>;
 pub type DroppedSlotsReceiver = Receiver<(Slot, BankId)>;
 
+/// interval to report 60s
+const BANK_DROP_SIGNAL_CHANNEL_REPORT_INTERVAL: u64 = 60_000;
+
+lazy_static! {
+    static ref BANK_DROP_LAST_REPORT_TIME: AtomicU64 = AtomicU64::new(0);
+}
+
 #[derive(Clone)]
 pub struct SendDroppedBankCallback {
     sender: DroppedSlotsSender,
 }
 
 impl DropCallback for SendDroppedBankCallback {
     fn callback(&self, bank: &Bank) {
-        if let Err(e) = self.sender.send((bank.slot(), bank.bank_id())) {
-            warn!("Error sending dropped banks: {:?}", e);
+        let ts = solana_sdk::timing::timestamp();
+
+        match self.sender.try_send((bank.slot(), bank.bank_id())) {
+            Err(TrySendError::Full(_)) => {
+                let last_report_time = BANK_DROP_LAST_REPORT_TIME.load(Ordering::Acquire);
+                if ts.saturating_sub(last_report_time) > BANK_DROP_SIGNAL_CHANNEL_REPORT_INTERVAL {
+                    inc_new_counter_info!("bank_drop_queue_full_event", 1);
```

Comment:
From this code: 

```
#[macro_export]
macro_rules! inc_new_counter_info {
    ($name:expr, $count:expr) => {{
        inc_new_counter!($name, $count, log::Level::Info, 0, 0);
    }};
    ($name:expr, $count:expr, $lograte:expr) => {{
        inc_new_counter!($name, $count, log::Level::Info, $lograte, 0);
    }};
    ($name:expr, $count:expr, $lograte:expr, $metricsrate:expr) => {{
        inc_new_counter!($name, $count, log::Level::Info, $lograte, $metricsrate);
    }};
}
```

I think `inc_new_counter_info` will automatically report a single metric with count `1` here.  For this reason we're starting to move away from `inc_new_counter_info` and just use `datapoint_info` with a `count` field, which is a lot more explicit.

From a functionality perspective, it's probably more useful to aggregate the number of calls that hit the buffer limit into a separate atomic `bank_drop_queue_full_event_count`, then compare and swap `0` when `BANK_DROP_SIGNAL_CHANNEL_REPORT_INTERVAL` is hit, , reporting a datapoint that represents the aggregate count of events in the last `BANK_DROP_SIGNAL_CHANNEL_REPORT_INTERVAL`

Reply:
Ah. you are right. Counter will only report the delta, not the accumulated value.
https://github.com/solana-labs/solana/blob/8eef3d9713db925a25965d6cbd8a0323a70598d1/metrics/src/counter.rs#L205
I will update to use datapoint instead.